index.html

```diff
@@ -179,44 +194,34 @@
-                            var villain = this;
-                            villain.bind('Moved', function(movement) {
-                                if (movement.axis == "x") {
-                                    if (villain.x - movement.oldValue < 0) {
-                                        villain._attackDirection = ATTACK_DIRECTIONS.LEFT;
-                                    } else {
-                                        villain._attackDirection = ATTACK_DIRECTIONS.RIGHT;
-                                    }
-                                } else {
-                                    if (villain.y - movement.oldValue < 0) {
-                                        villain._attackDirection = ATTACK_DIRECTIONS.UP;
-                                    } else {
-                                        villain._attackDirection = ATTACK_DIRECTIONS.DOWN;
-                                    }
-                                };
-                            });
-                            enemy.velocity() = 0;
+
+                        var x_distance = Math.abs(enemy.x - Crafty('PlayerCharacter').x);
+                        var y_distance = Math.abs(enemy.y - Crafty('PlayerCharacter').y);
+
+                        console.log(x_distance, y_distance);
+
+                        if (x_distance < 30 && y_distance < 30) {
+                            //enemy.velocity() = 0;
                             var attack = Crafty.e('EnemyAttack');
                             setTimeout (function() {
-                                switch(villain._attackDirection) {
+                                switch(enemy._attackDirection) {
                                     case ATTACK_DIRECTIONS.RIGHT:
-                                        attack.attr({x: villain.x + 30, y: villain.y})
+                                        attack.attr({x: enemy.x + 30, y: enemy.y})
                                         break;
                                     case ATTACK_DIRECTIONS.LEFT:
-                                        attack.attr({x: villain.x - 30, y: villain.y})
+                                        attack.attr({x: enemy.x - 30, y: enemy.y})
                                         break;
                                     case ATTACK_DIRECTIONS.UP:
-                                        attack.attr({x: villain.x, y: villain.y - 30})
+                                        attack.attr({x: enemy.x, y: enemy.y - 30})
                                         break;
                                     case ATTACK_DIRECTIONS.DOWN:
-                                        attack.attr({x: villain.x, y: villain.y + 30})
+                                        attack.attr({x: enemy.x, y: enemy.y + 30})
                                         break;
                                 };
-                            }, 50)
-                            setTimeout (function() {Crafty('EnemyAttack').destroy()}, 100)
-                    }, 10)
```

Comment:
this is debatable, but I got rid of the setInterval entirely and made it just event based (so on each movement event, the enemy checks all its things and updates whatever is necessary).

i'm not convinced this is the right event to bind to, but it should be event based. I would consider splitting this logic up a bit and having the "check what direction the enemy is attacking" bit be based on the "Move" event, and making another event that you fire outside handle the logic for when to attack. Then you could have a "game loop" under your main code, and every N milliseconds iterate through your enemies telling them to attack if capable etc.